app/gui2/src/providers/__tests__/widgetRegistry.test.ts

```diff
@@ -119,6 +120,33 @@ describe('WidgetRegistry', () => {
     expect(selectedFirst).toStrictEqual(widgetD)
     expect(selectedNext).toStrictEqual(widgetC)
   })
+
+  const widgetE = makeMockWidget(
+    'E',
+    defineWidget(WidgetInput.isAstOrPlaceholder, {
+      priority: -1,
+      score: () => Score.Perfect,
+      allowAsLeaf: false,
+    }),
+  )
+  test('perfect match does not allow being leaf', () => {
+    registry.registerWidgetModule(widgetE)
+    const case1 = registry.select(
+      { input: blankAst, nesting: 0 },
+      new Set([widgetA.default, widgetB.default, widgetC.default, widgetD.default]),
+    )
+    expect(case1).toBeUndefined
```

Comment:
Shouldn't it be a call?
```suggestion
    expect(case1).toBeUndefined()
```